src/cmap/message_stream.ts

```diff
@@ -191,7 +191,7 @@ function processIncomingData(stream: MessageStream, callback: Callback<Buffer>)
 
     if (messageBody.length !== messageHeader.length) {
       callback(
-        new MongoDriverError(
+        new MongoDecompressionError(
           'Decompressing a compressed message from the server failed. The message is corrupt.'
```

Comment:
Seems like we could improve the message a bit here, a MongoDecompressError implies the first sentence maybe its worth being specific, we can workshop but something like: 'Message header size does not match message body size'

Reply:
How does `'Message body and message header must be the same length'` sound?